Assets/MixedRealityToolkit.SDK/Inspectors/UX/Interactable/PressableButtonInspector.cs

```diff
@@ -61,120 +97,135 @@ private void OnSceneGUI()
+                return;
 
-            Vector3 touchCageSize = touchCage.size;
-            Vector3 touchCageCenter = touchCage.center;
-            Bounds touchCageLocalBounds = new Bounds(touchCageCenter, touchCageSize);
+            // If the button is being pressed, don't gather new info
+            // Just display the info we already gathered
+            // This lets people view button presses in real-time
+            if (button.IsTouching)
+            {
+                DrawButtonInfo(currentInfo, false);
+            }
+            else
+            {
+                currentInfo = GatherCurrentInfo();
+                DrawButtonInfo(currentInfo, EditingEnabled);
+            }
+        }
 
-            Transform buttonContentTransform = transform;
-            if (movingButtonVisuals.objectReferenceValue != null)
-                buttonContentTransform = (movingButtonVisuals.objectReferenceValue as GameObject).transform;
+        private ButtonInfo GatherCurrentInfo()
+        {
+            ButtonInfo info = new ButtonInfo();
 
-            // Get the start and end pos for moving content
-            Vector3 startPos = buttonContentTransform.localPosition;
-            Vector3 endPos = buttonContentTransform.localPosition;
-            endPos.z += maxPushDistance.floatValue;
+            info.touchCageLocalBounds = new Bounds(touchCage.center, touchCage.size);
             // Get the start pos for touch
-            Vector3 touchStartPos = touchCageLocalBounds.center;
-            touchStartPos.z -= touchCageLocalBounds.extents.z;
-
-            // This is where we'll store the results of our manipulations
-            float newMaxPushDistance = maxPushDistance.floatValue;
-            float newPressDistance = pressDistance.floatValue;
-            float newReleaseDistanceDelta = releaseDistanceDelta.floatValue;
-            Vector3 newStartPos = startPos;
-            Vector3 newEndPos = endPos;
-            Vector3 newPressDistPos = startPos + (Vector3.forward * newPressDistance) / transform.lossyScale.z;
-            Vector3 newTouchStartPos = touchStartPos;
-            Vector3 newReleaseDistPos = newPressDistPos + (Vector3.back * newReleaseDistanceDelta) / transform.lossyScale.z;
-            Vector3 newTouchCageSize = touchCageSize;
-            Vector3 newTouchCageCenter = touchCageCenter;
+            Vector3 touchStartOrigin = info.touchCageLocalBounds.center;
+            touchStartOrigin.z -= info.touchCageLocalBounds.extents.z;
+            info.touchStartOrigin = touchStartOrigin;
+            info.touchCageCenter = info.touchCageLocalBounds.center.z;
+            info.touchCageSize = info.touchCageLocalBounds.size.z;
+            
+            // Get the start and end pos for moving content
+            info.startPos = buttonContentTransform.localPosition.z;
+            info.endPos = buttonContentTransform.localPosition.z;
+            info.endPos += maxPushDistance.floatValue / transform.lossyScale.z;
```

Comment:
We should make sure that the info endPos will display correctly even if the scale of the pressablebutton is very small like 0.01. Isn't maxPushDistance given in world units?

Reply:
Good question - I thought it was in world space initially but without this fix the visual press distance didn't seem to match up with the press events on oddly scaled buttons. This is the issue we worked out over Teams. Can @danielhofmann-ms weigh in on this?

Reply:
Indeed `maxPushDistance` is currently in world space. But I thought all the distances in the info are in local space? So this makes sense.

On the other hand I just realized that lossyScale.z is the axis scaling in world space direction z, not local space z (press direction). So this only works with uniform scales. 

The scale along the local z direction can be computed like this:
`scale = transform.TransformVector(new Vector3(0, 0, 1)).magnitude`
Use `TransformVector` here to also apply scaling.

Reply:
I'm going to leave scaling as-is for this PR and incorporate TransformVector into the next round of fixes.